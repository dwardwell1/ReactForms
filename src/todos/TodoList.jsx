import {useState} from "react";
import {v4 as uuid} from "uuid"; 
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";


const TodoList = () => {
    const INITIAL_STATE = [
        {id:uuid(), todo:"Walk The Dog"},
        {id:uuid(), todo:"Take a Bath"}]
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo) => { 
        setTodos(todos => [...todos, {...newTodo, id:uuid()}]);
    }
    const removeTodo = (TodoId) => {
        setTodos(todos => todos.filter(t => t.id !== TodoId));
    }
    
    return (    
        <div>
            <h3>My Todo List</h3>
            <NewTodoForm addTodo={addTodo} />
            <div>
                <ul>
                {todos.map(({id,todo}) => (<li> <Todo id={id} todo={todo} removeTodo={removeTodo}/></li>))}
                </ul>
            </div>
        </div>
)}

export default TodoList;
