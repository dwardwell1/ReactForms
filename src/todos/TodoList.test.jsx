import {render, fireEvent, getByLabelText, findAllByText, getAllByText} from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function(){
    render(<TodoList />);
});

it("matches snapshot", function(){
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add item to shopping list", function(){
    const {queryByText, getByLabelText} = render(<TodoList />);
    const input = getByLabelText("New Todo");

    const btn = queryByText("Add Todo")
    expect(queryByText("Add Shrek")).not.toBeInTheDocument();
    fireEvent.change(input, {target: {value: "Add Shrek"}});
    fireEvent.click(btn);
    expect(queryByText("Add Shrek")).toBeInTheDocument();
   
});