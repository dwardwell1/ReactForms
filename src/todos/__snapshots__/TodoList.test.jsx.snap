// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      My Todo List
    </h3>
    <form>
      <label
        for="todo"
      >
        New Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="New Todo?"
        type="text"
        value=""
      />
      <input
        type="submit"
        value="Add Todo"
      />
    </form>
    <div>
      <ul>
        <li>
           
          <div>
            <h4>
               Walk The Dog  
              <button>
                 x 
              </button>
            </h4>
          </div>
        </li>
        <li>
           
          <div>
            <h4>
               Take a Bath  
              <button>
                 x 
              </button>
            </h4>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
